### PR TITLE
cluster: report size change conditions

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@etcd-bot retest this please

File: pkg/spec/cluster.go [hunk 1]
```diff
@@ -146,3 +146,44 @@ func (cs *ClusterStatus) SetVersion(v string) {
...
+		Type:           ClusterConditionReady,
+		TransitionTime: time.Now(),
+	}
+
+	if len(cs.Conditions) == 0 {
+		cs.appendCondition(c)
+		return
+	}
+
+	lastc := cs.Conditions[len(cs.Conditions)-1]
```
Comment (hongchaodeng) [inline pkg/spec/cluster.go]: It's unsafe to use this index here.
Maybe:
```go
var lastc *ClusterCondition
if le := len(cs.Conditions); le > 0 {
  lastc = &cs.Conditions[le-1]
}
if lastc != nil && ...
```

File: pkg/spec/cluster.go [hunk 2]
```diff
@@ -146,3 +146,44 @@ func (cs *ClusterStatus) SetVersion(v string) {
...
+
+func (cs *ClusterStatus) AppendScalingDownCondition() {
+	c := ClusterCondition{
+		Type:           ClusterConditionScalingDown,
+		TransitionTime: time.Now(),
+	}
+	cs.appendCondition(c)
+}
+
+func (cs *ClusterStatus) SetReadyCondition() {
```
Comment (hongchaodeng) [inline pkg/spec/cluster.go]: Why this is "Set" but other two "Append"?
Maybe change all of them "Set"?

Comment (xiang90) [inline pkg/spec/cluster.go]: They are different. Append appends the condition to the end of the list.

Set sets the last condition to the desired one. So it only appends if the last condition is not Ready; If the last condition is Ready, it does nothing.

File: pkg/spec/cluster.go (hunk 1 above)
Comment (xiang90) [inline pkg/spec/cluster.go]: not sure what you mean. i think line172 already does what you want?

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**